### PR TITLE
Connect parse

### PR DESCRIPTION
**Pull request by sophiabersani**
Adds integration for fetching data from remote Parse servers. Also handles creation of local servers for development purposes.

Additions for remote:
- Fetches Surveys
- Fetches Forms
- Fetches Questions
- Fetches Triggers

Additions for local:
- Creates Parse objects for testing Surveys, Forms, Questions, and Triggers.
- Adds a method to reset the server and erase data.
- Adds distinct iOS/Android configuration for local servers

Updates:
- Cleans Store from most dummy data. They have been replaced by Parse objects.
- Refactors SurveyListItem to function with Parse objects.

### CODE REVIEW

File: js/views/QuestionTypes/ShortAnswer.js [hunk 1]
```diff
@@ -15,7 +15,7 @@ const ShortAnswer = React.createClass ({
   render() {
     return (
       <View>
-        <Text style={Styles.form.questionLabel}>{this.props.question.text}</Text>
+        <Text style={Styles.type.h1}>{this.props.question.get('text')}</Text>
```
Comment (nathanathan) [inline js/views/QuestionTypes/ShortAnswer.js:18]: It looks like this style was accidentally changed when merging.

File: js/views/SurveyListPage.js [hunk 2]
```diff
@@ -30,9 +33,23 @@ const SurveyListPage = React.createClass ({
...
+      loadSurveyList({}, this.loadList)
   },
 
   /* Methods */
+  loadList(error, response){
+    if (error) {
+      console.warn(error)
+    } else {
+      this.setState({
+        isLoading: false,
```
Comment (nathanathan) [inline js/views/SurveyListPage.js:47]: It looks like this variable isn't used. If we do make use of it, it needs to be set to false when the Store already has surveys loaded.

File: js/api/Forms.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import _ from 'lodash'
+import Parse from 'parse/react-native'
```
Comment (nathanathan) [inline js/api/Forms.js:2]: package.json doesn't include the parse-react package I assume this is imported from.

File: js/api/ParseServer.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    default: connectToCustomServer(server); break;
+  }
+}
+
+function connectToLocalServer() {
+  Parse.initialize('testapp')
+  if (Platform.OS === 'android') {
+    Parse.serverURL = 'http://10.0.2.2:1337/parse'
+  } else {
+    Parse.serverURL = 'http://localhost:1337/parse'
```
Comment (dan-nyanko) [inline js/api/ParseServer.js:26]: Let's place this string in a highly visible area.  Maybe a settings.js file or use a package like this https://www.npmjs.com/package/react-native-config

File: js/api/PopulateLocalParseServer.js [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+      console.warn('Failed to create Survey, with error code: ' + error.message)
+    }
+  })
+}
+
+function createForms(parentSurvey) {
+  const Form = Parse.Object.extend('Form')
+  let newForm = new Form()
+
+  newForm.set('survey', parentSurvey)
```
Comment (jgoley) [inline js/api/PopulateLocalParseServer.js]: We are actually storing the relation on the survey class—surveys have many forms.
This is how we are doing it in the admin area: https://github.com/ecohealthalliance/mobile-survey/blob/pseudo-master/app/server/methods.coffee#L60-L62

File: js/api/PopulateLocalParseServer.js [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+
+function createQuestions(parentForm) {
+  const Question = Parse.Object.extend('Question')
+  for (var i = 0; i < DummyData.questions.length; i++) {
+    let newQuestion = new Question()
+    
+    newQuestion.set('text', DummyData.questions[i].text)
+    newQuestion.set('questionType', DummyData.questions[i].questionType)
+    newQuestion.set('properties', DummyData.questions[i].properties)
+    newQuestion.set('form', parentForm)
```
Comment (jgoley) [inline js/api/PopulateLocalParseServer.js]: Same here—storing the relation on forms.

File: js/api/PopulateLocalParseServer.js [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+  }
+}
+
+function createTriggers(parentForm) {
+  const Trigger = Parse.Object.extend('Trigger')
+  let newTrigger = new Trigger()
+
+  newTrigger.set('triggerType', DummyData.trigger.triggerType)
+  newTrigger.set('properties', DummyData.trigger.properties)
+  newTrigger.set('form', parentForm)
```
Comment (jgoley) [inline js/api/PopulateLocalParseServer.js]: Not setup yet on server/admin but this relation should also be stored on the form.

**Comment by sophiabersani** [comment]
Thanks Jon!
I've updated the PR to reflect that architecture.
This should be 1:1 for local development now.

**Pull request merged**